library/core/src/num/mod.rs

```diff
@@ -148,6 +154,33 @@ macro_rules! widening_impl {
         /// assert_eq!(5u32.carrying_mul(2, 10), (20, 0));
         /// assert_eq!(1_000_000_000u32.carrying_mul(10, 0), (1410065408, 2));
         /// assert_eq!(1_000_000_000u32.carrying_mul(10, 10), (1410065418, 2));
+        #[doc = concat!($AdaptiveTestPrefix, "assert_eq!(",
+            stringify!($SelfT), "::MAX.carrying_mul(", stringify!($SelfT), "::MAX, ", stringify!($SelfT), "::MAX), ",
+            "(0, ", stringify!($SelfT), "::MAX));"
+        )]
```

Comment:
`$AdaptiveTestPrefix` is a bit of a hack because this currently doesn't pass on signed types.  It's unclear to me whether it should -- there's open questions about this method on signed types in things like https://github.com/rust-lang/rust/issues/90493 -- so for now it's just skipped and hidden.